Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h

```diff
@@ -144,71 +187,144 @@
-  #define MISO_PIN                          PB14  // SPI2
-  #define MOSI_PIN                          PB15  // SPI2
-  #define SD_DETECT_PIN                     PF12  // SD_CD
-#else
-  // SD as custom software SPI (SDIO pins)
-  #define SCK_PIN                           PC12
-  #define MISO_PIN                          PC8
-  #define MOSI_PIN                          PD2
-  #define SS_PIN                            -1
-  #define ONBOARD_SD_CS_PIN                 PC11
-  #define SDSS                              PD2
-  #define SD_DETECT_PIN                     -1
 #endif
 
+//
+// SPI
+//
+#define ENABLE_SPI2
+#define SCK_PIN                             PB13
+#define MISO_PIN                            PB14
+#define MOSI_PIN                            PB15
+// SPI2 is already shared by LCD touch driver and flash
+// SPI2 MOSI (PB15) can be retrieved by soldering job on R17
+// SPI1(PA7) & SPI3(PB5) are not available
+
+//
+// Trinamic TMC2208/2209 UART
+//
 #if HAS_TMC_UART
+
+  // Reduce baud rate for software serial reliability
+  #if HAS_TMC_SW_SERIAL
+    #define TMC_BAUD_RATE                  19200
+  #endif
+
   /**
    * TMC2208/TMC2209 stepper drivers
    *
    * Hardware serial communication ports.
    * If undefined software serial is used according to the pins below
    */
-  //#define X_HARDWARE_SERIAL  MSerial1
-  //#define Y_HARDWARE_SERIAL  MSerial1
-  //#define Z_HARDWARE_SERIAL  MSerial1
-  //#define E0_HARDWARE_SERIAL MSerial1
-  //#define E1_HARDWARE_SERIAL MSerial1
-
-  // Unused servo pins may be repurposed with SoftwareSerialM
-  //#define X_SERIAL_TX_PIN                 PF8   // SERVO3_PIN -- XS2 - 6
-  //#define Y_SERIAL_TX_PIN                 PF9   // SERVO2_PIN -- XS2 - 5
-  //#define Z_SERIAL_TX_PIN                 PA1   // SERVO1_PIN -- XS1 - 6
-  //#define E0_SERIAL_TX_PIN                PC3   // SERVO0_PIN -- XS1 - 5
-  //#define X_SERIAL_RX_PIN      X_SERIAL_TX_PIN
-  //#define Y_SERIAL_RX_PIN      Y_SERIAL_TX_PIN
-  //#define Z_SERIAL_RX_PIN      Z_SERIAL_TX_PIN
-  //#define E0_SERIAL_RX_PIN    E0_SERIAL_TX_PIN
 
-  // Reduce baud rate for software serial reliability
-  #if HAS_TMC_SW_SERIAL
-    #define TMC_BAUD_RATE                  19200
+  // Hardware serial on Serial 1
+  #define TMC_HARDWARE_SERIAL
+  #if ENABLED(TMC_HARDWARE_SERIAL)
+    #define X_HARDWARE_SERIAL            Serial1
+    #define X2_HARDWARE_SERIAL           Serial1
+    #define Y_HARDWARE_SERIAL            Serial1
+    #define Y2_HARDWARE_SERIAL           Serial1
+    #define Z_HARDWARE_SERIAL            Serial1
+    #define Z2_HARDWARE_SERIAL           Serial1
+    #define E0_HARDWARE_SERIAL           Serial1
+    #define E1_HARDWARE_SERIAL           Serial1
+    #define E2_HARDWARE_SERIAL           Serial1
+    #define E3_HARDWARE_SERIAL           Serial1
+    #define E4_HARDWARE_SERIAL           Serial1
   #endif
+
+  // Software serial on unused servo pins
+  //#define TMC_SOFTWARE_SERIAL
+  #if ENABLED(TMC_SOFTWARE_SERIAL)
```

Comment:
You need to set TMC_BAUD_RATE to 19200 in the Software Serial case. The default baud rate will not work, at least not reliably.